kong/globalpatches.lua

```diff
@@ -288,35 +288,36 @@ return function(options)
     local old_tcp_connect
     local old_udp_setpeername
 
-    local function tcp_resolve_connect(sock, host, port, sock_opts)
-      local target_ip, target_port = toip(host, port)
-      if not target_ip then
-        return nil, "[toip() name lookup failed]: " .. tostring(target_port) -- err
+  -- need to do the extra check here: https://github.com/openresty/lua-nginx-module/issues/860
+    local function strip_nils(port, opts)
+      if port and opts then
+        return port, opts
+      elseif opts then
+        return opts
```

Comment:
here you are "shifting" a parameter? is that what's intended? So in case of a unix socket, the proper call would be `tcp_connect(sock, name, opts)` instead of `tcp_connect(sock, name, nil, opts)`?

Reply:
and you are correct, sorry for the noise. https://github.com/openresty/lua-nginx-module#tcpsockconnect

Reply:
*Since the function should be compatible with the ngx.socket.tcp calls, in the case of a unix socket, the caller would do connect(sock, name, opts), so in the code, the opts table would ba carried in the port variable actually.*

isn't that what will happens? if the `port` is nil then `strip_nils` will return only the `opts`,
so `return f(sock, target_ip, strip_nils(target_port, sock_opts))` equals `return f(sock, target_ip, sock_opts)` when `target_port` is nil

Reply:
oh i got what you means, sorry

Reply:
since the third argument can be port or opts (and will be confusing), can i do it like this?

```lua
    local function strip_nils(first, second)
      if second then
        return first, second
      elseif first then
        return first
      end
    end

    local function resolve_connect(f, sock, host, port_or_opts, opts)
      if sub(host, 1, 5) == "unix:" then
        opts = port_or_opts -- in unix domain socket, third argument are socket options
        return f(sock, host, strip_nils(opts))
      end

      local port = port_or_opts
      local target_ip, target_port = toip(host, port)
      if not target_ip then
        return nil, "[toip() name lookup failed]: " .. tostring(target_port)
      end
      return f(sock, target_ip, strip_nils(target_port, opts))
    end

    local function tcp_resolve_connect(sock, host, port_or_opts, opts)
      return resolve_connect(old_tcp_connect, sock, host, port_or_opts, opts)
    end

    local function udp_resolve_setpeername(sock, host, port)
      return resolve_connect(old_udp_setpeername, sock, host, port)
    end
```

Reply:
yes, seems appropriate